### PR TITLE
Fixing find help message and behavior

### PR DESCRIPTION
**Pull request by harshavardhana**

### CODE REVIEW

**Comment by harshavardhana** [comment]
The changes and wording in the docs were approved by @abperiasamy while reviewing the `find` command help message.

File: cmd/find-main.go [hunk 1]
```diff
@@ -123,37 +123,34 @@ FORMAT
       {url} --> Substitutes to a shareable URL of the path.
 
 EXAMPLES:
-   01. Find all files named foo from all buckets.
+   01. ind all "foo.jpg" from all buckets from "s3" account
```
Comment (nitisht) [inline cmd/find-main.go]: s/ind/Find

File: cmd/find-main.go [hunk 2]
```diff
@@ -123,37 +123,34 @@ FORMAT
...
        $ {{.HelpName}} s3 --name "file"
```
Comment (nitisht) [inline cmd/find-main.go]: Shouldn't the name be `foo.jpg` here

File: cmd/find-main.go [hunk 3]
```diff
@@ -123,37 +123,34 @@ FORMAT
...
-   07. Find all jpeg images from "s3/bucket" and copy to minio "play/bucket" *continuously*.
+   06. Find all jpg images from "s3/bucket" and copy to minio "play/bucket" *continuously*.
        $ {{.HelpName}} s3/bucket --name "*.jpg" --watch --exec "mc cp {} play/bucket"
 
-   08. Generate self expiring urls (7 days), for all objects between 64 MB, and 1 GB in size.
+   07. Generate public URLs valid for 7 days, for all objects between 64 MB, and 1 GB in size.
        $ {{.HelpName}} s3 --larger 64MB --smaller 1GB --print {url}
 
-   09. Find all files under the s3 bucket which were created within a week.
+   08. Find all objects created within last week.
```
Comment (nitisht) [inline cmd/find-main.go]: created _during_ last week?

Comment (harshavardhana) [reply cmd/find-main.go]: No within .. during is not right here IMO

File: cmd/find-main.go (hunk 2 above)
Comment (harshavardhana) [reply cmd/find-main.go]: Yes mistake

File: cmd/find-main.go (hunk 3 above)
Comment (nitisht) [reply cmd/find-main.go]: :+1:

File: cmd/find-main.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/find-main.go]: Done

File: cmd/find-main.go [hunk 4]
```diff
@@ -123,37 +123,34 @@ FORMAT
...
 
-   04. Copy all jpeg files from AWS S3 "s3/photos" bucket to minio "play/photos" bucket.
-       $ {{.HelpName}} s3/photos --name "*.jpg" --exec "mc cp {} play/photos"
-
-   05. Find all jpeg images from any folder prefixed with "album".
+   04. Find all jpg images from any folder prefixed with "album".
        $ {{.HelpName}} s3/photos --name "*.jpg" --path "*/album*/*"
 
-   06. Find all files with extensions jpeg, png, and gif using regex.
+   05. Find all objects with extensions jpg, png, and gif using regex.
```
Comment (ebozduman) [inline cmd/find-main.go]: `Find all objects with extensions jpg, png, and gif using regex.`
 =>
`Find all objects with extensions ".jpg", ".png", and ".gif" using regex under "s3/photos".`

File: cmd/find-main.go [hunk 5]
```diff
@@ -123,37 +123,34 @@ FORMAT
...
        $ {{.HelpName}} s3/photos --regex "(?i)\.(jpg|png|gif)$"
 
-   07. Find all jpeg images from "s3/bucket" and copy to minio "play/bucket" *continuously*.
+   06. Find all jpg images from "s3/bucket" and copy to minio "play/bucket" *continuously*.
```
Comment (ebozduman) [inline cmd/find-main.go]: `Find all jpg images from "s3/bucket" and copy to minio "play/bucket" *continuously*.`
 =. 
`Find all objects with ".jpg" extension under "s3/bucket" and copy to minio "play/bucket" *continuously*.`

File: cmd/find-main.go [hunk 6]
```diff
@@ -123,37 +123,34 @@ FORMAT
...
        $ {{.HelpName}} s3/bucket --name "*.jpg" --watch --exec "mc cp {} play/bucket"
 
-   08. Generate self expiring urls (7 days), for all objects between 64 MB, and 1 GB in size.
+   07. Generate public URLs valid for 7 days, for all objects between 64 MB, and 1 GB in size.
```
Comment (ebozduman) [inline cmd/find-main.go]: `and 1 GB in size.`
 =>
`and 1 GB in size under "s3".`

File: cmd/find-main.go [hunk 7]
```diff
@@ -123,37 +123,34 @@ FORMAT
...
        $ {{.HelpName}} s3 --larger 64MB --smaller 1GB --print {url}
 
-   09. Find all files under the s3 bucket which were created within a week.
+   08. Find all objects created in last week.
```
Comment (ebozduman) [inline cmd/find-main.go]: `Find all objects created in last week.`
 =>
`Find all objects created in the last week under "s3/bucket".`

File: cmd/find-main.go [hunk 8]
```diff
@@ -123,37 +123,34 @@ FORMAT
...
        $ {{.HelpName}} s3/bucket --newer 1w
 
-   10. Find all files which were created more than 6 months ago ignoring files ending in jpg.
+   09. Find all objects which were created more than 6 months ago, additionally ignore all jpg objects.
```
Comment (ebozduman) [inline cmd/find-main.go]: `Find all objects which were created more than 6 months ago, additionally ignore all jpg objects.`
 =>
`Find all objects created more than 6 months ago, and exclude the ones with ".jpg" extension under "s3".`

File: cmd/find-main.go [hunk 9]
```diff
@@ -123,37 +123,34 @@ FORMAT
...
        $ {{.HelpName}} s3 --older 6m --ignore "*.jpg"
 
-   11. List all objects up to 3 levels subdirectory deep.
+   10. List all objects up to 3 levels sub-directory deep.
```
Comment (ebozduman) [inline cmd/find-main.go]: `List all objects up to 3 levels sub-directory deep.`
 =>
`List all objects up to 3 levels sub-directory deep under "s3/bucket".`

File: cmd/find-main.go (hunk 5 above)
Comment (harshavardhana) [reply cmd/find-main.go]: It kept on purpose to indicate that they are images, we don't need to say just extension.

File: cmd/find.go [hunk 10]
```diff
@@ -218,11 +218,13 @@ func find(ctx *findContext, fileContent contentMessage) {
...
 	} // For all matching content
+
 	// proceed to either exec, format the output string.
-	switch {
-	case ctx.execCmd != "":
+	if ctx.execCmd != "" {
 		execFind(stringsReplace(ctx.execCmd, fileContent))
-	case ctx.printFmt != "":
+		return
+	}
```
[the remark this answers is not in the document]
Comment (harshavardhana) [reply cmd/find.go]: We want to change the original behavior @ebozduman

**Comment by harshavardhana** [comment]
@ebozduman - PTAL

**Pull request merged**